### ISSUE TITLE
Improved support for GitHub Enterprise 3.1 and up

### ISSUE THREAD

**Issue by alexr00** [microsoft/vscode-pull-request-github#3983]
In September, we explored whether we could improve support for GitHub Enterprise. 

TL;DR: Auth issues were fixed and now GitHub Enterprise is supported, starting with GitHub Enterprise version 3.1.

Details of the investigation:

- There were issues around auth and landing users in a dead end. These are resolved.
- Users often got stuck creating a PAT. Thanks to @TylerLeonhardt we have PAT-less auth for enterprise and several users have already confirmed that this solves their issues.
- The core scenarios that I tested work all the way back to GHES version 3.1, which is deprecated. 3.2 is the oldest non-deprecated version. Because of this, no versioning story is needed at present for GHPRI 🎉
- We can easily detect the GHES version as it's returned in a header from the first request we make to check if the server is github. However, we're not currently using this due to the previous point.

Open question:

- I'm fairly certain that 3.0 doesn't work with GHPRI due to use of PR auto-merge API. Since the oldest supported version is 3.2, I'm not sure if it's worth adding versioning just to support this.
